backend/platform_settings_v2/platform_auth_service.py

```diff
@@ -0,0 +1,253 @@
+
+logger = logging.getLogger(__name__)
+
+
+class PlatformAuthenticationService:
+    """Service class to hold Platform service authentication and validation.
+
+    Supports generation, refresh, revoke and toggle of active keys.
+    """
+
+    @staticmethod
+    def generate_platform_key(
+        is_active: bool,
+        key_name: str,
+        user: User,
+        organization: Optional[Organization] = None,
+    ) -> dict[str, Any]:
+        """Method to support generation of new platform key. Throws error when
+        maximum count is exceeded. Forbids for user other than admin
+        permission.
+
+        Args:
+            key_name (str): Value of the key
+            is_active (bool): By default the key is False
+            user (User): User object representing the user generating the key
+            organization (Optional[Organization], optional):
+                Org the key belongs to. Defaults to None.
+
+        Returns:
+            dict[str, Any]:
+                A dictionary containing the generated platform key details,
+                    including the id, key name, and key value.
+        Raises:
+            DuplicateData: If a platform key with the same key name
+                already exists for the organization.
+            InternalServiceError: If an internal error occurs while
+                generating the platform key.
+        """
+        organization: Organization = organization or UserContext.get_organization()
+        if not organization:
+            raise InternalServiceError("No valid organization provided")
+        try:
+            # TODO : Add encryption to Platform keys
+            # id is added here to avoid passing of keys in transactions.
+            platform_key: PlatformKey = PlatformKey(
+                id=str(uuid.uuid4()),
+                key=str(uuid.uuid4()),
+                is_active=is_active,
+                organization=organization,
+                key_name=key_name,
+                created_by=user,
+                modified_by=user,
+            )
+            platform_key.save()
+            result: dict[str, Any] = {}
+            result[PlatformServiceConstants.ID] = platform_key.id
+            result[PlatformServiceConstants.KEY_NAME] = platform_key.key_name
+            result[PlatformServiceConstants.KEY] = platform_key.key
+
+            logger.info(f"platform_key is generated for {organization.id}")
+            return result
+        except IntegrityError as error:
+            logger.error(f"Integrity error - failed to generate platform key : {error}")
+            raise DuplicateData(
+                f"{ErrorMessage.KEY_EXIST}, \
+                            {ErrorMessage.DUPLICATE_API}"
+            )
+        except Exception as error:
+            logger.error(f"Failed to generate platform key : {error}")
+            raise InternalServiceError()
+
+    @staticmethod
+    def delete_platform_key(id: str) -> None:
```

Comment:
@muhammad-ali-e should we include other details like org or something as well to make sure the key deletions happens only for the respected org? So we could even modify the query to include org? Will that make sense here?

Reply:
We can add it in logs without accepting the params. Can be done by later. Anyway the Id here is primary key, So I think it not needed here for an extra validation.

Reply:
The only reason to suggest the org also to be passed is to make sure we delete the PK of indented org alone even if we are deleting using the primarykey/

May be we can add a TODO for revisit this later

Reply:
Added a TODO comment here